src/app/core/cache/builders/build-decorators.ts

```diff
@@ -1,23 +1,41 @@
 import 'reflect-metadata';
 
 import { GenericConstructor } from '../../shared/generic-constructor';
-import { CacheableObject } from '../object-cache.reducer';
+import { CacheableObject, TypedObject } from '../object-cache.reducer';
 import { ResourceType } from '../../shared/resource-type';
 
 const mapsToMetadataKey = Symbol('mapsTo');
 const relationshipKey = Symbol('relationship');
 
 const relationshipMap = new Map();
+const typeMap = new Map();
 
-export function mapsTo(value: GenericConstructor<CacheableObject>) {
-  return Reflect.metadata(mapsToMetadataKey, value);
+export function mapsTo(value: GenericConstructor<TypedObject>) {
```

Comment:
Could we add some TypeDocs to these updated/refactored functions (all the refactored ones in this file)?  These are the key functions you are refactoring in this PR, so this seems like a good opportunity to better document them via TypeDocs.